From 702042f4c6b4e820cad69aff3fcfeaeff662706f Mon Sep 17 00:00:00 2001
From: Maruthi Srinivas Bayyavarapu <maruthi.srinivas.bayyavarapu@xilinx.com>
Date: Wed, 5 Dec 2018 13:00:01 +0530
Subject: [PATCH 0136/1851] ASoC: xlnx: Fixes in audio formatter driver

commit 0e90b635acf19e786343eb5a4bfd6b00adbfa0f4 from
https://github.com/Xilinx/linux-xlnx.git

Two changes done in this cleanup patch:
1. Memory for object is freed after all references to it are done.
2. Function 'xlnx_formatter_pcm_reset' is made static.

Signed-off-by: Maruthi Srinivas Bayyavarapu <maruthi.srinivas.bayyavarapu@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 sound/soc/xilinx/xlnx_formatter_pcm.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/sound/soc/xilinx/xlnx_formatter_pcm.c b/sound/soc/xilinx/xlnx_formatter_pcm.c
index a0637b58f036..221a52b7f18c 100644
--- a/sound/soc/xilinx/xlnx_formatter_pcm.c
+++ b/sound/soc/xilinx/xlnx_formatter_pcm.c
@@ -310,7 +310,7 @@ static int parse_consumer_format(u32 chsts_reg1_val, u32 chsts_reg2_val,
 	return 0;
 }
 
-void xlnx_formatter_pcm_reset(void __iomem *mmio_base)
+static void xlnx_formatter_pcm_reset(void __iomem *mmio_base)
 {
 	u32 val;
 
@@ -443,11 +443,10 @@ static int xlnx_formatter_pcm_close(struct snd_pcm_substream *substream)
 	val &= ~AUD_CTRL_IOC_IRQ_MASK;
 	writel(val, stream_data->mmio + XLNX_AUD_CTRL);
 
-	kfree(stream_data);
-
 	if (substream->stream == SNDRV_PCM_STREAM_CAPTURE)
 		xlnx_formatter_pcm_reset(stream_data->mmio + XLNX_S2MM_OFFSET);
 
+	kfree(stream_data);
 	return 0;
 }
 
-- 
2.31.1

